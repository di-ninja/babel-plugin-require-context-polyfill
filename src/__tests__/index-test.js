const babel = require('@babel/core')
const plugin = require('../')

it('sould require dirA files', () => {
  const {code} = babel.transformFileSync('fixtures/testA.js', {plugins: [plugin]})
  expect(code).toMatchSnapshot()
})

it('sould require root project dirs', () => {
  const {code} = babel.transformFileSync('fixtures/testB.js', {plugins: [plugin]})
  expect(code).toMatchSnapshot()
})
