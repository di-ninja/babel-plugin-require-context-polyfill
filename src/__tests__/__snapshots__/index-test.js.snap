// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sould require dirA files 1`] = `
"const foo =
/*!@compileDependencies([\\"./dirA/\\"])*/
function () {
  const map = Object.defineProperties({}, {
    \\"hello.js\\": {
      value: require(\\"fixtures/dirA/hello.js\\"),
      enumerable: true
    }
  });

  const returnContext = function (item) {
    return map[item];
  };

  returnContext.keys = function () {
    return Object.keys(map);
  };

  return returnContext;
}();"
`;

exports[`sould require root project dirs 1`] = `
"const foo =
/*!@compileDependencies([\\".//\\"])*/
function () {
  const map = Object.defineProperties({}, {
    \\"dirA\\": {
      value: require(\\"fixtures/dirA\\"),
      enumerable: true
    },
    \\"testA.js\\": {
      value: require(\\"fixtures/testA.js\\"),
      enumerable: true
    },
    \\"testB.js\\": {
      value: require(\\"fixtures/testB.js\\"),
      enumerable: true
    }
  });

  const returnContext = function (item) {
    return map[item];
  };

  returnContext.keys = function () {
    return Object.keys(map);
  };

  return returnContext;
}();"
`;
